portals/publisher/src/main/webapp/source/src/app/components/Scopes/EditScope.jsx

```diff
@@ -242,10 +242,25 @@ class EditScope extends React.Component {
         const promise = APIValidation.role.validate(base64url.encode(role));
         promise
             .then(() => {
-                this.setState({
-                    roleValidity: true,
-                    validRoles: [...validRoles, role],
-                });
+                const splitRole = role.split('/', 2);
+                let validatedRole = '';
+                if (splitRole.length > 1) {
+                    const domain = splitRole.length > 0 ? splitRole[0] : '';
+                    if (domain.toUpperCase() !== 'INTERNAL') {
+                        const domainUpperCase = domain.toUpperCase().concat('/');
+                        validatedRole = domainUpperCase.concat(splitRole[1]);
+                    } else {
+                        validatedRole = role;
+                    }
+                } else {
+                    validatedRole = role;
+                }
+                if (!validRoles.includes(validatedRole)) {
+                    this.setState({
+                        roleValidity: true,
+                        validRoles: [...validRoles, validatedRole],
+                    });
+                }
```

Comment:
Can we implement this code segment in a single place and reuse that in these 4 places? Possibly in Shared Component since these are different components?

Reply:
Lets fix this in another PR as this is an improvement @shilmyhasan Please note